### PR TITLE
MOC Filigree Vector

### PR DESCRIPTION
**Pull request by Simisays**
Wording is kinda weird but I think this is correct ?

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/filigree_vector.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Name:Filigree Vector
+ManaCost:3 W
+Types:Artifact Creature Phyrexian Construct
+PT:1/1
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigCounter | TriggerDescription$ When CARDNAME enters the battlefield, put a +1/+1 counter on each of any number of target creatures and a charge counter on each of any number of target artifacts.
+SVar:TrigCounter:DB$ PutCounter | ValidTgts$ Creature | TgtPrompt$ Select any number of target creatures to distribute +1/+1 counters to | CounterType$ P1P1 | CounterNum$ X | TargetMin$ 0 | TargetMax$ X | DividedAsYouChoose$ X | SubAbility$ DBChargeCounter
+SVar:DBChargeCounter:DB$ PutCounter | ValidTgts$ Artifact | TgtPrompt$ Select any number of target artifacts to distribute charge counters to | CounterType$ CHARGE | CounterNum$ Y | TargetMin$ 0 | TargetMax$ Y | DividedAsYouChoose$ Y
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/filigree_vector.txt:7]: It's not divided as you choose if it doesn't say so
You just target and put counters on them 🤷

**Review by tool4ever — DISMISSED** [review]

File: forge-gui/res/cardsfolder/upcoming/filigree_vector.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Name:Filigree Vector
+ManaCost:3 W
+Types:Artifact Creature Phyrexian Construct
+PT:1/1
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigCounter | TriggerDescription$ When CARDNAME enters the battlefield, put a +1/+1 counter on each of any number of target creatures and a charge counter on each of any number of target artifacts.
+SVar:TrigCounter:DB$ PutCounter | ValidTgts$ Creature | TgtPrompt$ Select any number of target creatures to distribute +1/+1 counters to | CounterType$ P1P1 | CounterNum$ 1 | TargetMin$ 0 | TargetMax$ X | SubAbility$ DBChargeCounter
+SVar:DBChargeCounter:DB$ PutCounter | ValidTgts$ Artifact | TgtPrompt$ Select any number of target artifacts to distribute charge counters to | CounterType$ CHARGE | CounterNum$ 1 | TargetMin$ 0 | TargetMax$ Y
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/upcoming/filigree_vector.txt:7]: i need to check the rules if both counters are put on them at the same time for cards that trigger of counters

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/filigree_vector.txt:7]: i didn't explicit checked rules yet, but my gut is telling me that both types of counters needs to be put on the cards at the same time

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/filigree_vector.txt:7]: So the current version needs to be changed or is it correct ?

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/filigree_vector.txt:7]: Right now you've got 2 PutCounter lines, so your P1P1 counters are put first (possibly triggering "CounterAdded" triggers) then your CHARGE counters are put...

I feel like we may have something like this wired up already, but I'm not sure.

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/filigree_vector.txt:7]: Brokers Ascendancy and a few Ajani PW cards use the params you want, but in a non targeted way. I think you could rig this with a few pump lines for targeting and some remembering (maybe imprinting), then a PutCounterAll line with ValidCards2/CounterType2

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/filigree_vector.txt:7]: Thanks I'll give it a try

File: forge-gui/res/cardsfolder/upcoming/filigree_vector.txt [hunk 3]
```diff
@@ -3,8 +3,10 @@ ManaCost:3 W
...
 PT:1/1
 T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigCounter | TriggerDescription$ When CARDNAME enters the battlefield, put a +1/+1 counter on each of any number of target creatures and a charge counter on each of any number of target artifacts.
-SVar:TrigCounter:DB$ PutCounter | ValidTgts$ Creature | TgtPrompt$ Select any number of target creatures to distribute +1/+1 counters to | CounterType$ P1P1 | CounterNum$ 1 | TargetMin$ 0 | TargetMax$ X | SubAbility$ DBChargeCounter
-SVar:DBChargeCounter:DB$ PutCounter | ValidTgts$ Artifact | TgtPrompt$ Select any number of target artifacts to distribute charge counters to | CounterType$ CHARGE | CounterNum$ 1 | TargetMin$ 0 | TargetMax$ Y
+SVar:TrigCounter:DB$ Pump | ValidTgts$ Creature | TgtPrompt$ Select any number of target creatures to distribute +1/+1 counters to | TargetMin$ 0 | TargetMax$ X | RememberTargets$ True | SubAbility$ DBChargeCounter
+SVar:DBChargeCounter:DB$ Pump | ValidTgts$ Artifact | TgtPrompt$ Select any number of target artifacts to distribute charge counters to | TargetMin$ 0 | TargetMax$ Y | ImprintCards$ ThisTargetedCard | SubAbility$ DBPutCounterAll
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/filigree_vector.txt:7]: Give this a try–
Use dev mode to give AI a few creatures, an artifact creature, and an artifact.
Then use dev mode to make the AI cast this card...
If the AI doesn't give its stuff counters (or starts giving opponent stuff counters yikes), try adding `AILogic$ Pump` to each of these lines and try again.

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/filigree_vector.txt:7]: Yeah AI put a charge counter on my Black Lotus

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/filigree_vector.txt]: Does the `AILogic$ Pump` seem to make it behave better?

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/filigree_vector.txt]: AI still giving charge counters to my artifacts 
image

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/filigree_vector.txt]: I guess AI removedeck:Random ?

File: forge-gui/res/cardsfolder/upcoming/filigree_vector.txt [hunk 4]
```diff
@@ -0,0 +1,17 @@
+Name:Filigree Vector
+ManaCost:3 W
+Types:Artifact Creature Phyrexian Construct
+PT:1/1
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigCounter | TriggerDescription$ When CARDNAME enters the battlefield, put a +1/+1 counter on each of any number of target creatures and a charge counter on each of any number of target artifacts.
+SVar:TrigCounter:DB$ Pump | ValidTgts$ Creature | AILogic$ Pump | TgtPrompt$ Select any number of target creatures to distribute +1/+1 counters to | TargetMin$ 0 | TargetMax$ X | RememberTargets$ True | SubAbility$ DBChargeCounter
+SVar:DBChargeCounter:DB$ Pump | ValidTgts$ Artifact | AILogic$ Pump | TgtPrompt$ Select any number of target artifacts to distribute charge counters to | TargetMin$ 0 | TargetMax$ Y | ImprintCards$ ThisTargetedCard | SubAbility$ DBPutCounterAll
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/filigree_vector.txt:7]: Without digging too deep into the PumpAi my guess is it isn't wired up handle "Pump" on artifacts...

Maybe try one more thing before we put the RemoveDeck labels on there... add `AITgts$ Artifact.YouCtrl`

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/filigree_vector.txt:7]: Oh seems to work now. I'll keep AITgts$ in mind for the future
image

**Review by Northmoc — APPROVED** [review]

**Pull request merged**